### DISCUSSION TITLE
How to cache RetrieveModelMixin with persistent cache

### DISCUSSION THREAD

**Discussion by mfoglio** [encode/django-rest-framework#8521] [Question & Answer]
I have the following Django REST view:
```
class AnnotationViewSet(
        mixins.CreateModelMixin,
        mixins.ListModelMixin,
        mixins.RetrieveModelMixin,
        viewsets.GenericViewSet
    ):
    queryset = Annotation.objects.all()
    serializer_class = AnnotationSerializer
```
Accessing a single method will call the `retrieve` function thanks to `mixins.RetrieveModelMixin`. I would like to speed up the function since it requires multiple queries and it uses a lot of CPU.  
In particular:
- I'd like a persistent cache that could be used after restarting the application
- The cached data needs to be different based on the record retrieved (e.g. http://127.0.0.1/annotations/1 vs http://127.0.0.1/annotations/2, etc)

As of now I tried to overwrite the view `retrieve` method: 
```
    @method_decorator(cache_page(60 * 60 * 24 * 365))
    def retrieve(self, request, *args, **kwargs):
        instance = self.get_object()
        serializer = self.get_serializer(instance)
        return Response(serializer.data)
```
and to set the default cache to disk:
```
CACHES = {
    'default': {
        'BACKEND': 'django.core.cache.backends.filebased.FileBasedCache',
        'LOCATION': '/home/django_cache',
    }
}
```
I don't think it is working, as I need to cache about 2 million items (I have about 2 millions records): the folder `/home/django_cache` uses less than a Mb and it's size changes every time I call the API.

**Comment by yyyyyyyan** [comment]
Hi! The view is cached when a user access the URL. So if no one requested the URL before, the results won't be cached yet.
